components/bitbucket/new-commit.js

```diff
@@ -0,0 +1,152 @@
+const bitbucket = require("https://github.com/PipedreamHQ/pipedream/components/bitbucket/bitbucket.app.js");
+
+module.exports = {
+  name: "New Commit (Instant)",
+  description: "Emits an event when a new commit is pushed to a branch",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    bitbucket,
+    db: "$.service.db",
+    http: "$.interface.http",
+    workspaceId: { propDefinition: [bitbucket, "workspaceId"] },
+    repositoryId: {
+      propDefinition: [
+        bitbucket,
+        "repositoryId",
+        c => ({ workspaceId: c.workspaceId }),
+      ],
+    },
+    branchName: {
```

Comment:
Can we make this prop optional?

Reply:
(Maybe should even be `branchNames` but that might be not as convenient as it should punt on supporting 0-n branches for now -- but 0 or 1 might be a nice improvement for now?)

Reply:
The approach here is the same as with Gitlab and Github, so in the spirit of consistency we should do the same here.
(We can always make a required argument optional, but not the other way around without breaking backward compatibility).

However, I agree that it would be nicer to have it as an optional parameter, and I don't think it'd require that much effort other than testing multiple cases.
I can refactor all 3 event sources (i.e. Github, Gitlab and BitBucket) in a separate PR if that's OK.

Reply:
Added a task for this: https://github.com/PipedreamHQ/pipedream/issues/508